web/src/lib/chat/fetchSomeChatData.ts

```diff
@@ -0,0 +1,236 @@
+import {
+  AuthTypeMetadata,
+  getAuthTypeMetadataSS,
+  getCurrentUserSS,
+} from "@/lib/userSS";
+import { fetchSS } from "@/lib/utilsSS";
+import {
+  CCPairBasicInfo,
+  DocumentSet,
+  Tag,
+  User,
+  ValidSources,
+} from "@/lib/types";
+import { ChatSession } from "@/app/chat/interfaces";
+import { Persona } from "@/app/admin/assistants/interfaces";
+import { InputPrompt } from "@/app/admin/prompt-library/interfaces";
+import { FullEmbeddingModelResponse } from "@/components/embedding/interfaces";
+import { Settings } from "@/app/admin/settings/interfaces";
+import { fetchLLMProvidersSS } from "@/lib/llm/fetchLLMs";
+import { LLMProviderDescriptor } from "@/app/admin/configuration/llm/interfaces";
+import { Folder } from "@/app/chat/folders/interfaces";
+import { personaComparator } from "@/app/admin/assistants/lib";
+import { cookies } from "next/headers";
+import {
+  SIDEBAR_TOGGLED_COOKIE_NAME,
+  DOCUMENT_SIDEBAR_WIDTH_COOKIE_NAME,
+} from "@/components/resizable/constants";
+import { hasCompletedWelcomeFlowSS } from "@/components/initialSetup/welcome/WelcomeModalWrapper";
+import { fetchAssistantsSS } from "../assistants/fetchAssistantsSS";
+import { NEXT_PUBLIC_DEFAULT_SIDEBAR_OPEN } from "../constants";
+
+interface FetchChatDataResult {
+  user?: User | null;
+  chatSessions?: ChatSession[];
+  ccPairs?: CCPairBasicInfo[];
+  availableSources?: ValidSources[];
+  documentSets?: DocumentSet[];
+  assistants?: Persona[];
+  tags?: Tag[];
+  llmProviders?: LLMProviderDescriptor[];
+  folders?: Folder[];
+  openedFolders?: Record<string, boolean>;
+  defaultAssistantId?: number;
+  toggleSidebar?: boolean;
+  finalDocumentSidebarInitialWidth?: number;
+  shouldShowWelcomeModal?: boolean;
+  shouldDisplaySourcesIncompleteModal?: boolean;
+  userInputPrompts?: InputPrompt[];
+}
+
+type FetchOption =
+  | "user"
+  | "chatSessions"
+  | "ccPairs"
+  | "documentSets"
+  | "assistants"
+  | "tags"
+  | "llmProviders"
+  | "folders"
+  | "userInputPrompts";
+
+/* 
+NOTE: currently unused, but leaving here for future use. 
+*/
+export async function fetchSomeChatData(
```

Comment:
Not used, but in the future could be used for pages like AssistantGallery which don't need everything